test/core/linter_test.py

```diff
@@ -585,3 +589,16 @@ def test_delayed_exception():
     de = runner.DelayedException(ve)
     with pytest.raises(ValueError):
         de.reraise()
+
+
+@pytest.mark.here
+def test__attempt_to_change_templater_warning(caplog):
+    """Test warning if user tries to change templater in .sqlfluff file in subdirectory"""
+    initial_config = FluffConfig(configs={"core": {"templater": "jinja"}})
+    lntr = Linter(config=initial_config)
+    updated_config = FluffConfig(configs={"core": {"templater": "dbt"}})
+    with caplog.at_level(logging.WARNING):
+        rendered_string = lntr.render_string(
+            in_str="select * from table", fname="test.sql", config=updated_config
+        )
+    assert f"Attempt to set templater to {updated_config.get('templater_obj').name} failed. in caplog.text"
```

Comment:
@dmateusp I didn't explicitly write and use new .sqlfluff and subdirectory fixtures but this tests the same functionality. Is this OK or do you think it is worth setting up some new fixtures?